src/main/java/com/adyen/httpclient/HttpURLConnectionClient.java

```diff
@@ -81,7 +101,13 @@ public String request(String requestUrl, String requestBody, Config config, bool
 
         setContentType(httpConnection, APPLICATION_JSON_TYPE);
 
-        return doPostRequest(httpConnection, requestBody);
+        String response = doPostRequest(httpConnection, requestBody);
+
+        if (config.getSkipCertificationValidation()) {
+            installDefaultTrustManager();
```

Comment:
Before releasing the lib this can be modified using the correct terminal certificate